README.md
# Tutellus.io
The landing page, kyc process and dashboard for the Tutellus ICO.

## Development
;TL;DR
- requires node v8
- install via `npm i`
- edit environment file `.env`.
- run with `npm start`

This is a create-react-app, the original instructions, development scripts and
related documentation are detailed in DEVELOPERS.md.

This project uses some environment variables during build time to configure the
final bundle. Those can be found with `git grep process.env src/` in order to
create a .env file containing the required. The specifics (for overriding) are
detailed in DEVELOPERS.md, an example `.env` file would look like:
```
REACT_APP_FBTRACKERID=facebook tracker code
REACT_APP_MAILCHIMP_URL=url for mailchimp forms (see mailchimp dashboard)
REACT_APP_MAILLIST_GENERAL=mailing list id for general subscribers (from landing page)
REACT_APP_MAILLIST_REGISTER=mailing list id for registered users (for registration)
REACT_APP_FIREBASE_DATABASEURL=firebase url (see firebase docs)
REACT_APP_FIREBASE_APIKEY=api key for configured firebase db
REACT_APP_FIREBASE_AUTHDOMAIN=auth domain for firebase
REACT_APP_FIREBASE_PROJECTID=firebase project identifier
REACT_APP_FIREBASE_STORAGEBUCKET=firebase storage identifier for KYC documents uploading
REACT_APP_FIREBASE_MESSAGINGSENDERID=used to send registration mails, passwd recovery, etc.
```

### Contributing
The guidelines will be detailed in CONTRIBUTING.md, refer to it for any specifics.

Contribute your changes via a pull request, without adding lint or flowtype
errors.

Use the issue tracker to request features and bugfixes. Check prior issues and
fixes to learn about the project and decissions made.

## Production and deployment
The full application is a SPA statically served, generated via `npm run build`.
Probably any HTTP server would do.

The production configuration can override any keys the local .env file as detailed
in DEVELOPERS.md, for instance `.env.production`: 
```
REACT_APP_FIREBASE_DATABASEURL=production firebase instance
REACT_APP_FIREBASE_APIKEY=production application key
#etc..
```

If the production file is present it will override the local config. Is therefore
recommended to either control wich file is present prior to the build or
sourcing the appropiate config to avoid issues (the latter aproach should work
but hasn't been tested yet).


CHANGELOG.md
### 3.0.4 (2018-10-02)
- feat(Banner) adding tokens bubble and video on top (e513c81)
- refactor(GTMetrics) externalizing the images (f94ff00)

### 3.0.3 (2018-10-01)
- feat(i18n) update IDN language (a3e71d1)

### 3.0.2 (2018-10-01)
- fix(i18n) deleting some documents (197455d)

### 3.0.1 (2018-10-01)
- fix(Crowdsale) adding height settings to donut chart (a4d32bd)
- refactor(optimice) change index js location to improve performance. (5631955)

## 3.0.0 (2018-09-28)
- Javier Ortiz - refactor(react-sizeme) updating library version (cf848a9)
- Javier Ortiz - refactor(Timer) delete unused component (66cb776)
- Javier Ortiz - refactor(lifecicle) changing methods react lifecycle (c9e16e4)
- Javier Ortiz - fix(audits) best practices chrome audits (f53f61f)
- Javier Ortiz - fix(npm) Fix registry to registry.npmjs.org (08d8af3)
- Javier Ortiz - fix(nodejs) updating node and package versions (f3f5905)
- Javier Ortiz - feat(STO) first version of the home page (ffb7d18)
- Javier Ortiz - feat(script) new script to serve static. Use: npm run serve (ee794c2)
- Javier Ortiz - refactor(flow-types) updating flow-types (2a707b1)
- Javier Ortiz - feat(Home) cleaning components that will no longer be used in the home (59c4c0e)
- Javier Ortiz - refactor(i18n) deleted commented lines (0d3a390)
- Javier Ortiz - feat(CryptonomosReferral) delete CryptonomosReferral (f3be0b4)

### 2.5.12 (2018-09-18)
- feat(CountDown) hiding the timer (b4016c8)

### 2.5.11 (2018-08-02)
- feat(IE) Show modal, must use another browser css:grid not supported (782e441)
- fix(IE) adding polyfills for new methods (70e98ac)
- fix(Firefox) delete rel="preload" for font.css. onload event don't work (62e4caf)

### 2.5.10 (2018-07-11)
- fix(Crowdsale) hide preico table info (4d3b845)

### 2.5.9 (2018-07-11)
- fix(crowdsale) change image token distribution (41380e2)
- fix(i18n) new document in korean; info about roadshow in Seoul (676e4b7)

### 2.5.8 (2018-07-05)
- fix(i18n) Update Crowdsale table to new Hard Cap (4a397fb)
- feat(ICOPurchases) Omit ICOPurchases while retail is shy (be5598f)

### 2.5.7 (2018-06-28)
- fix(i18n) update crowdsale dates (23cb3d7)

### 2.5.6 (2018-06-25)
- fix(i18n) add chinese praises (5e62c0c)

### 2.5.5 (2018-06-15)
- fix(CryptonomosReferral) sending credentials on fetch (13daf43)

### 2.5.4 (2018-06-15)
- feat(modal) New copies suggested by Cryptonomos (d89c10c)

### 2.5.3 (2018-06-14)
- feat(modal) added modal with cryptonomos process explained (979ce94)

### 2.5.2 (2018-06-14)
- feat(i18n) add new advisor (fc6f8f2)

### 2.5.1 (2018-06-13)
- fix(styles) fixes for mobile layout (40fced1)
- feat(CountDown) Check page visibility for check server time. (3f2c21d)
- feat(visibility) Check page visibility for disable purchase polling (2d0db97)

## 2.5.0 (2018-06-12)
- feat(sticky) added sticky bar with countdown and contribute button (57f2b25)

### 2.4.8 (2018-06-08)
- feat(roadshow) Adding roadshow events (2064f76)
- fix(suscription) fixed config mailchimp (550f93b)

### 2.4.7 (2018-06-07)
- feat(cryptonomos) Adding referral snippet code. (49bbf91)

### 2.4.6 (2018-06-07)
- feat(analytics) Tracking click events (9ad2e23)
- fix(unmount) Adding unmount method to remove listeners/intervals (9bf9244)

### 2.4.5 (2018-06-06)
- feat(ratings) adding carousel with autoplay (1ad4328)
- fix(imageCarousel) LazyLoad using react-slick (8caac792)

### 2.4.4 (2018-06-05)
-  feat(ratings) Adding top ratings sites (4d2f7b4)

### 2.4.3 (2018-06-05)
- fix(i18n) fixes (747ffdc)

### 2.4.2 (2018-06-05)
- feat(telegram) Adding local communities telegram groups (d84cf7f)
- fix(analytics) Use withTracker with named tracker. Cryptonomos adds a default tracker that disables the one we were using. (d387d68)

### 2.4.1 (2018-06-04)
- feat(i8n) New languages AR and VI. New Praises added. (480da34)
- feat(facebook) Adding cryptonomos facebook pixel (a432d2b)
- fix(toast) Show 3 last transactions (1f7f2d6)

## 2.4.0 (2018-05-31)
- feat(toast) Adding toast with the latest ICO purchases (e738574)
- fix(eslint) Linter fixes (d46fef1)

### 2.3.1 (2018-05-31)
- fix(i18n) PH language fixes (8c5c5ba)

## 2.3.0 (2018-05-30)
- feat(i18n) ID and PH langs added (094533f)
- feat(lang-selector) New lang selector with flags (601a764)

### 2.2.4 (2018-05-28)
-  feat(i18n) adding FR language (7e5ec4a)

### 2.2.3 (2018-05-24)
- fix(i18n) update min PreSale purchase to 0.05ETH (91dee95)

### 2.2.2 (2018-05-24)
- feat(intro) New countdown with bonus info (cf9a645)
- fix(i18n) little fixes on DE (d0984fa)

### 2.2.1 (2018-05-22)
- feat(i18n) Translated to DE. Promo 1ETH till June 12. (9673a62)

## 2.2.0 (2018-05-16)
- feat(notify-bar) added notify bar with "1ETH till 20 may" active promo (f72d4b1)

### 2.1.10 (2018-05-11)
- fix(i18n) RU updated thanx to Cryptonomos (7ebd39c)

### 2.1.9 (2018-05-10)
- feat(countdown) Setting time to main sale June 12, 10 am CET (8c71f35)

### 2.1.8 (2018-05-10)
- feat(join) Adding Buy button under timer (7809183)
- fix(timer) when loading with past time shows 0 (298a67b)

### 2.1.7 (2018-05-10)
- fix(i18n) Updated presale time (a44675f)
- Revert "feat(countdown) Setting time to main sale" (2018c9b)

### 2.1.6 (2018-05-10)
- feat(countdown) Setting time to main sale (2584da7)

### 2.1.5 (2018-05-09)
- feat(crowdsale) new rows with TUT price and ICO bonuses (51122cb)
- feat(footer) email contact added, fixed link terms of token sale (7b95724)

### 2.1.4 (2018-05-07)
- fix(i18n) Update countries disclaimer (9793081)

### 2.1.3 (2018-05-03)
- feat(join ico) Adding countries disclaimer (29275bd)


src/App.js
//@flow
// $FlowFixMe: usamos moment/moment para que funcione el transpilado (se podría hacer con webpack)
import moment from 'moment/moment';
import React, {Component} from 'react';
import {injectGlobal} from 'styled-components';
import {I18n} from 'react-i18next';
import i18n from './i18n';

import styled from 'styled-components';
import {
    MainHeader,
    MAIN_HEADER_HEIGHT,
    MainMenu,
    PageBanner,
    PageTitle,
    SectionImage,
    Text,
    PageSection,
    Row,
    Col,
    SectionTitle,
    PlayButton,
    CTAButton,
    Features,
    Feature,
    Quote,
    Team,
    TeamMember,
    CenteredImage,
    CrowdsaleTable,
    Roadmap,
    Milestone,
    FAQ,
    CenteredBlock,
    PageFooter,
    FooterBranding,
    FooterNav,
    NavLink,
    NavCategory,
} from './components';
import styles from './styles';

const Benefits = styled.div`
    display:grid;
    grid-template-columns: repeat(2, 50%);
`;
const UserGroup = styled(props =>
    <div className={ props.className }>
        <h4>{ props.name }</h4>
        <div>{ props.children }</div>
    </div>
)`
    display: grid;
    grid-column-gap: 10px;
    grid-template-columns: 33% 67%;
    align-items: center;

    & > h4 {
        text-align: center;
        font-weight: bold;
        text-transform: uppercase;

        &:before {
            content: '';
            background: url(${ props => props.icon }) center center no-repeat;
            padding: 2em;
            display: block;
            margin-bottom: 1em;
        }
    }
`;
const BulletList = styled.ul``;
const BulletPoint = styled.li`
    line-height: 2em;
    &:before {
        content: '';
        display: inline-block;
        border: solid 5px transparent;
        border-left-color: ${ styles.colors.lightblue };
        border-right: none;
        margin: 0 10px;
    }
`;

const SOCIAL_LINKS = {
    facebook: '//fb.me/tutellus',
    twitter: '//twitter.com/tutellus',
    linkedin: '//linkedin.com/tutellus',
};
class WindowScroll extends React.Component {
    constructor() {
        super();
        this.state = {x: 0, y: 0};
    }
    componentDidMount() {
        window.addEventListener('scroll', evt => {
            this.setState({x: window.scrollX, y: window.scrollY});
        });
    }
    render() {
        return this.props.children(this.state);
    }
};

class App extends Component/*::<{}>*/ {
    componentDidMount() {
        injectGlobal`${ styles.global }`;
    }
    render() {
        return (
            <I18n i18n={ i18n }>{ (t, {i18n}) =>
                <div>
                    <WindowScroll>{ scroll =>
                        <MainHeader small={ scroll.y > MAIN_HEADER_HEIGHT } logo="/images/white-logo.svg">
                            <MainMenu onLanguage={ lang => i18n.changeLanguage(lang) } socialLinks={ SOCIAL_LINKS } locale={ i18n.language } />
                        </MainHeader>
                    }</WindowScroll>
                    <main>
                        <PageBanner dark backgroundVideo="/images/bgvideo.mp4">
                            <PageTitle dangerouslySetInnerHTML={{ __html: t("page_title") }} />
                            <Text center dangerouslySetInnerHTML={{ __html: t('page_subtitle') }} />
                            <PlayButton video="/images/bgvideo.mp4" />
                        </PageBanner>
                        <PageSection>
                            <SectionTitle dangerouslySetInnerHTML={{ __html: t('the_pain_title') }} />
                            <Text center>{ t('the_pain') }</Text>
                            <Row>
                                <Col size={ 1 / 2}>
                                    <SectionImage src="/images/edtech.svg" />
                                </Col>
                                <Col size={ 1 / 2}>
                                    <SectionImage src="/images/itjobs.svg" />
                                </Col>
                            </Row>
                        </PageSection>
                        <PageSection light interstitialImage="/images/tokens.png">
                            <SectionTitle dangerouslySetInnerHTML={{ __html: t('the_solution_title') }} />
                            <Text center>{ t('the_solution') }</Text>
                        </PageSection>
                        <PageSection title={ t('benefits_title') }>
                            <Text center>{ t('benefits_description') }</Text>
                            <Benefits>
                                <UserGroup name={ t('students') } icon="/images/students.svg">
                                    <BulletList>
                                        <BulletPoint>{ t('benefit_earn_studying') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_learn_more') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_relevance') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_use_crypto') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_wordwide_payments') }</BulletPoint>
                                    </BulletList>
                                </UserGroup>
                                <UserGroup name={ t('teachers') } icon="/images/teachers.svg">
                                    <BulletList>
                                        <BulletPoint>{ t('benefit_instant_payments') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_earn_via_best_students') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_earn_via_relevance') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_earn_via_subscriptions') }</BulletPoint>
                                        <BulletPoint>{ t('benefit_exclusive_services') }</BulletPoint>
                                    </BulletList>
                                </UserGroup>
                            </Benefits>
                        </PageSection>
                        <PageSection dark title={ t('how_it_works_title') } image="/images/HTTW.png">
                            <Text center>{ t('how_it_works') }</Text>
                            <CenteredBlock>
                                <CTAButton icon="http://placehold.it/20x20" primary>{ t('download_whitepaper') }</CTAButton>
                            </CenteredBlock>
                        </PageSection>
                        <PageSection title={ t('platform_title') } image="https://www.tutellus.com/bower_components/tutellus.css/images/home/desktop-computer.jpg">
                            <Features>
                                <Feature title="1.000.000" image="/images/students.svg">
                                    { t('students') }
                                </Feature>
                                <Feature title="+2.000.000" image="/images/transactions.svg">
                                    { t('transactions') }
                                </Feature>
                                <Feature title="+1b min" image="/images/learning.svg">
                                    { t('learning') }
                                </Feature>
                                <Feature title="$10m" image="/images/transactioned.svg">
                                    { t('transactioned') }
                                </Feature>
                            </Features>
                        </PageSection>
                        <PageSection light>
                            <Quote href="https://elpais.com/economia/2015/07/10/actualidad/1436521761_125197.html" cite="El Pais, 2016">
                                { t('the_leading_platform_quote') }
                            </Quote>
                            <CenteredBlock>
                                <CTAButton secondary>{ t('goto_tutellus') }</CTAButton>
                            </CenteredBlock>
                        </PageSection>
                        <PageSection title={ t('team_title') }>
                            <Team>
                                <TeamMember name="Miguel Caballero" title="CEO" photo="https://www.tutellus.com/bower_components/tutellus.css/images/team/Miguel.jpg">
                                    { t('mcaballero_bio') }
                                </TeamMember>
                                <TeamMember name="Javier Ortiz" title="CTO" photo="https://www.tutellus.com/bower_components/tutellus.css/images/team/JaviSok.jpg">
                                    { t('jortiz_bio') }
                                </TeamMember>
                                <TeamMember name="Carlos López" title="Engineer" photo="https://www.tutellus.com/bower_components/tutellus.css/images/team/Carlos.jpg">
                                    { t('clopez_bio') }
                                </TeamMember>
                                <TeamMember name="Javier Calvo" title="Data Scientist" photo="https://www.tutellus.com/bower_components/tutellus.css/images/team/JaviMat.jpg">
                                    { t('jcalvo_bio') }
                                </TeamMember>
                                <TeamMember name="Karol Szymanczak" title="Designer" photo="https://www.tutellus.com/bower_components/tutellus.css/images/team/Karol.jpg">
                                    { t('karol_bio') }
                                </TeamMember>
                            </Team>
                        </PageSection>
                        <PageSection dark title={ t('crowdsale_title') }>
                            <Row>
                                <Col size={ 1 / 2 }>
                                    <CenteredImage src="https://placehold.it/800x600" />
                                </Col>
                                <Col size={ 1 / 2 }>
                                    <CenteredImage src="https://placehold.it/800x600" />
                                </Col>
                            </Row>
                            <CrowdsaleTable>
                                <tbody>
                                    <tr>
                                        <td>{ t('token_name') }</td><td>TUT</td>
                                    </tr>
                                    <tr>
                                        <td>{ t('preico_date') }</td><td>{ t('preico_date_value') }</td>
                                    </tr>
                                    <tr>
                                        <td>{ t('ico_date') }</td><td>{ t('ico_date_value') }</td>
                                    </tr>
                                    <tr>
                                        <td>{ t('total_supply') }</td><td>{ t('total_supply_value') }</td>
                                    </tr>
                                    <tr>
                                        <td>{ t('amount_for_sale') }</td><td>{ t('amount_for_sale_value') }</td>
                                    </tr>
                                    <tr>
                                        <td>{ t('soft_cap') }</td><td>{ t('soft_cap_value') }</td>
                                    </tr>
                                    <tr>
                                        <td>{ t('accepted_currencies') }</td><td>ETH</td>
                                    </tr>
                                </tbody>
                            </CrowdsaleTable>
                            <CenteredBlock>
                                <CTAButton secondary>{ t('register_for_the_crowdsale') }</CTAButton>
                            </CenteredBlock>
                        </PageSection>
                        <PageSection title={ t("roadmap_title") }>
                            <Text center>{ t("roadmap_description") }</Text>
                            <Roadmap>
                                <Milestone done title={ t("mvp") } date={ moment('2014-01') }>
                                    <ul>
                                        <li>{ t("usercount_100k") }</li>
                                        <li>{ t("coursecount_20k") }</li>
                                    </ul>
                                </Milestone>
                                <Milestone done title={ t("new_platform") } date={ moment('2015-05') }>
                                    <ul>
                                        <li>{ t("api_plus_mean_fw") }</li>
                                        <li>{ t("usercount_300k") }</li>
                                        <li>{ t("coursecount_40k") }</li>
                                    </ul>
                                </Milestone>
                                <Milestone done title={ t("crowdsale") } date={ moment('2017-11') } />
                                <Milestone title={ t("ico") } date={ moment('2018-01') } />
                                <Milestone title={ t("core_features") } date={ moment('2018-06') }>
                                    <ul>
                                        <li>{ t("wallets") }</li>
                                        <li>{ t("token_interop") }</li>
                                    </ul>
                                </Milestone>
                                <Milestone title={ t("value_services") } date={ moment('2018-12') }>
                                    <ul>
                                        <li>{ t("full_offchain_interop") }</li>
                                    </ul>
                                </Milestone>
                                <Milestone title={ t("third_entities") } date={ moment("2019-06") }>
                                    <ul>
                                        <li>{ t("entities_design") }</li>
                                        <li>{ t("access_management") }</li>
                                    </ul>
                                </Milestone>
                                <Milestone title={ t("value_features") } date={ moment("2019-12") }>
                                    <ul>
                                        <li>{ t("product_management") }</li>
                                        <li>{ t("full_integration") }</li>
                                    </ul>
                                </Milestone>
                                <Milestone title={ t("api_features") } date={ moment("2020-06") }/>
                            </Roadmap>
                        </PageSection>
                        <PageSection light title={ t('Technology') }>
                            <Text center>{ t('technology_description') }</Text>
                        </PageSection>
                        <PageSection dark title={ t("FAQs") }>
                            <FAQ title={ t("are_there_restrictions") }>{ t('are_there_restrictions_answer') }</FAQ>

                            <FAQ title={ t("how_to_purchase") }>{ t('how_to_purchase_answer') }</FAQ>

                            <FAQ title={ t("nothing_happens") }>
                                <p>{ t('nothing_happens_window_dissapears') }</p>
                                <p>{ t('nothing_happens_check_connection') }</p>
                            </FAQ>

                            <FAQ title={ t("sent_twice") }>{ t('sent_twice_answer') }</FAQ>

                            <FAQ title="can_buy_from_exchanges">{ t('can_buy_from_exchanges_answer') }</FAQ>

                            <FAQ title={ t("is_erc20") }>{ t('is_erc20_answer') }</FAQ>

                            <FAQ title={ t("cant_find_in_exchanges") }>{ t('cant_find_in_exchanges_answer') }</FAQ>

                            <FAQ title={ t("marked_as_pending") }>
                                <ol>
                                    <li>{ t('check_steps') }</li>
                                    <li>{ t('check_txhash') }</li>
                                    <li>{ t('wait_a_few_hours') }</li>
                                    <li>{ t('contact_support') }</li>
                                </ol>
                            </FAQ>
                            <FAQ title={ t("why_its_promising") }>
                                <ol>
                                    <li>{ t('running_since_2013') }</li>
                                    <li>{ t('big_ecosystem') }</li>
                                    <li>{ t('proved_how_to_create_value') }</li>
                                    <li>{ t('already_many_contributors') }</li>
                                    <li>{ t('great_team') }</li>
                                </ol>
                            </FAQ>

                            <FAQ title={ t("what_is_tutellus") }>{ t('what_is_tutellus_answer') }</FAQ>

                            <FAQ title={ t("why_the_ico") }>{ t('why_the_ico_answer') }</FAQ>

                            <FAQ title={ t("benefits_for_tokenholders") }>{ t('benefits_for_tokenholders_answer') }</FAQ>

                            <FAQ title={ t( "advantages_vs_other_projects" ) }>
                                <p>{ t('big_community') }</p>
                                <p>{ t('solves_problems_for_all_players') }</p>
                                <p>{ t('natural_way_to_evolve') }</p>
                                <p>{ t('leading_in_latin_countries') }</p>
                            </FAQ>

                            <FAQ title={ t("is_tutellus_safe") }>
                                <p>{ t('never_been_hacked') }</p>
                                <p>{ t('secure_infrastructure') }</p>
                            </FAQ>
                        </PageSection>
                    </main>
                    {/* TODO: Falta el footer */}
                    <PageFooter>
                        <FooterBranding logo="/images/color-logo.svg" about={ t('about_tutellus') } socialLinks={ SOCIAL_LINKS } />
                        <FooterNav>
                            <NavCategory title={ t('company') }>
                                <NavLink>Tutellus</NavLink>
                                <NavLink>Platform</NavLink>
                                <NavLink>Roadmap</NavLink>
                                <NavLink>Terms of Use</NavLink>
                            </NavCategory>
                            <NavCategory title={ t('ICO') }>
                                <NavLink>Whitepaper</NavLink>
                                <NavLink>Token Sale</NavLink>
                                <NavLink>Wallet</NavLink>
                                <NavLink>Blog</NavLink>
                                <NavLink>Help</NavLink>
                            </NavCategory>
                            <NavCategory title={ t('CONTACT') }>
                                <NavLink href="mailto:ico@tutellus.com">ico@tutellus.com</NavLink>
                                <NavLink href="//2tel.us/tutellus-address" target="_blank">c/ Henri Dunant, 15. Madrid 28050.</NavLink>
                                <NavLink href="tel://+34910052511">+34 91 00 525 11</NavLink>
                            </NavCategory>
                        </FooterNav>
                    </PageFooter>
                </div>
            }</I18n>
        );
    }
}

export default App;
